tests/integration/components/registries/finalize-registration-modal/component-test.ts

```diff
@@ -102,4 +111,67 @@ module('Integration | Component | finalize-registration-modal', hooks => {
         // // Close the dialog
         this.set('isOpen', false);
     });
+
+    test('almost done modal content: no moderation', async function(assert) {
+        this.store = this.owner.lookup('service:store');
+        const noModerationProvider = server.create('registration-provider', { reviewsWorkflow: null });
+        const noModRegistration = server.create('registration', { provider: noModerationProvider });
+
+        const registrationModel = await this.store.findRecord('registration', noModRegistration.id);
+        this.set('draftManager', { provider: noModerationProvider });
+        this.set('model', registrationModel);
+        this.set('isOpen', true);
+        await render(hbs`
+            <Registries::FinalizeRegistrationModal::Manager
+                @registration={{this.model}}
+                @draftManager={{this.draftManager}}
+                as |manager|
+            >
+                <Registries::FinalizeRegistrationModal @isOpen={{this.isOpen}} @manager={{manager}} />
+            </Registries::FinalizeRegistrationModal::Manager>
+        `);
+        // Click immediate radio button
+        await click('[data-test-immediate-button]');
+        // Click submit button
+        await click('[data-test-submit-registration-button]');
+
+        const opts = { learnMoreLink: 'aaa.aa', htmlSafe: true };
+        assert.dom('[data-test-finalize-main]').hasTextContaining(
+            stripHtmlTags(t('registries.finalizeRegistrationModal.notice.noModerationFromProject', opts).toString()),
+            'modal shows warning',
+        );
+        assert.dom('[data-test-finalize-main]').doesNotHaveTextContaining(
+            'A moderator must review and approve', 'modal does not mention moderation for unmoderated providers',
+        );
+    });
+
+    test('almost done modal content: with moderation', async function(assert) {
+        this.store = this.owner.lookup('service:store');
+        const withModerationProvider = server.create('registration-provider');
```

Comment:
minor:
```suggestion
        const withModerationProvider = server.create('registration-provider', { reviewsWorkflow: 'pre-moderation' });
```

Reply:
`reviewsWorkflow` is set to 'pre-moderation' by default in the registration provider factory, so I don't think it's necessary to set it here. (May be best not to set the `reviewsWorkflow` in the factory, but that's a slightly different topic)